GNUmakefile

```diff
@@ -92,7 +94,10 @@ pkg/linux_arm64/nomad: CC = aarch64-linux-gnu-gcc
 endif
 
 ifeq (Darwin,$(THIS_OS))
-pkg/linux_%/nomad: CGO_ENABLED = 0
+pkg/linux_%/nomad:
+CGO_ENABLED = 0
+CGO_CFLAGS=-Wno-undef-prefix
+SDKROOT=$(xcrun --sdk macosx --show-sdk-path)
```

Comment:
The issue described in https://github.com/hashicorp/nomad/issues/11539 was for `make dev`, which will build for the OS you're running the build on. So shouldn't this be on the `pkg/darwin_%/nomad` target? If not and it impacts all builds from Darwin, I think the flags should be set outside the `pkg/linux_%/nomad` target.

Reply:
Yes, sorry.  Early morning eyes!

Reply:
Actually, I didn't create this `pkg/darwin_%/nomad` target, I just appended to it.  Are you saying it was an existent issue thatI should change?  I think the answer is yes, but just want to double check.

Reply:
It's a little subtle, but this appended to the `pkg/linux_%/nomad` target when building on Darwin. So this target is only for cross-compiling from Darwin to Linux. (Which breaks the `exec` driver as we know, but should at least compile for convenience).

So what I'm suggesting is that you want to add an explicit target for `pkg/darwin_%/nomad` that has these flags set. And that _maybe_ they need to be set for this cross-compiling target as well but I'm not sure.

Reply:
Ahhh ok.  Thanks.